vsintegration/src/FSharp.VS.FSI/sessions.fs

```diff
@@ -274,19 +275,46 @@ type FsiSession(sourceFile: string) =
         sprintf "FSIChannel_%d_%d_%d" pid tick salt
 
     let procInfo, fsiSupportsServer = fsiStartInfo channelName sourceFile
+
+    let usingNetCore = SessionsProperties.fsiUseNetCore
+
     let cmdProcess = new Process(StartInfo=procInfo)
     let fsiOutput = Event<_>()
     let fsiError = Event<_>()
 
     do cmdProcess.Start() |> ignore
 
+    let mutable cmdProcessPid = cmdProcess.Id
+    let mutable trueProcessPid = if usingNetCore then None else Some cmdProcessPid
+    let trueProcessIdFile = Path.GetTempFileName() + ".pid"
+
+    let mutable seenPidJunkOutput = false
+    let mutable skipLines = 0
+
     // hook up stdout\stderr data events
-    do readLinesAsync cmdProcess.StandardOutput (catchAll fsiOutput.Trigger)
+    do readLinesAsync cmdProcess.StandardOutput (fun line ->
+           // For .NET Core, the "dotnet fsi ..." starts a second process "dotnet ..../fsi.dll ..."
+           // So the first thing we ask a .NET Core F# Interactive to do is report its true process ID.
+           // 
+           // After it executes the request it will print:
+           //    LINE1 -->  val it: unit = ()
+           //    LINE2 -->  
+           //    LINE3 -->  SERVER-PROMPT>
+           //
+           // We skip these lines.
+           if usingNetCore && not seenPidJunkOutput && line.StartsWith("val ") then
```

Comment:
This is used to filter the output that arises from getting the process ID from the F# Interactive session